### ISSUE TITLE
Consider Using XLED Library

### ISSUE THREAD

**Issue by zanix** [dr1rrb/ha-twinkly#5]
There is an existing unofficial XLED library that could be used as a base for this component
https://github.com/scrool/xled

**Comment by dr1rrb** [comment]
Yeah I discovered it when I published my client to pypi in order to merge this integration to the home-assistant core repo (https://github.com/home-assistant/core/pull/42103). 

I've choose to not change it immediately so the integration could be merge before Christmas (even if unfortunately we missed the last release ... it should be in 0.119 which is planned for the 13/12), but the plan now is to remove my poor client and use 'xled' instead.

BTW: I'm not familiar with python so each feature takes me a while, and I'm really busy currently. So if anybody wants to do that, you're more than welcome :)

**Comment by zanix** [comment]
Oh nice, I didn't even notice the PR for integration into core.
Good luck on the refactor later on!

**Comment by zanix** [comment]
It looks like there was a previous attempt for a Home Assistant component by the XLED author themselves
https://github.com/home-assistant/core/pull/19158

**Comment by dr1rrb** [comment]
Ooooohhhh never saw this one. Then I should probably contact him ... he is probably also interested to have a greater integration in HA 😃

**Comment by jschlyter** [comment]
See also [TTLS](https://travis-ci.com/github/jschlyter/ttls) for a async version (as mentioned in https://github.com/home-assistant/core/pull/42103)
